File: exercise/c#/Day04/Day04.Tests/ArticleTests.cs
using FluentAssertions;
using Xunit;

namespace Day04.Tests
{
    public class ArticleTests
    {
        [Fact]
        public void It_Should_Add_Valid_Comment()
        {
            var article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
            );

            article.AddComment("Amazing article !!!", "Pablo Escobar");
        }

        [Fact]
        public void It_Should_Add_A_Comment_With_The_Given_Text()
        {
            var article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
            );

            const string text = "Amazing article !!!";
            article.AddComment(text, "Pablo Escobar");

            article.Comments
                .Should().HaveCount(1)
                .And.ContainSingle(comment => comment.Text == text);
        }

        [Fact]
        public void It_Should_Add_A_Comment_With_The_Given_Author()
        {
            var article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
            );

            const string author = "Pablo Escobar";
            article.AddComment("Amazing article !!!", author);

            article.Comments
                .Should().HaveCount(1)
                .And.ContainSingle(comment => comment.Author == author);
        }

        [Fact]
        public void It_Should_Add_A_Comment_With_The_Date_Of_The_Day()
        {
            var article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
            );

            article.AddComment("Amazing article !!!", "Pablo Escobar");

            var today = DateOnly.FromDateTime(DateTime.Today);
            article.Comments
                .Should().ContainSingle(comment => comment.CreationDate == today);
        }

        [Fact]
        public void It_Should_Throw_An_Exception_When_Adding_Existing_Comment()
        {
            var article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
            );
            article.AddComment("Amazing article !!!", "Pablo Escobar");

            var act = () => article.AddComment("Amazing article !!!", "Pablo Escobar");
            act.Should().Throw<CommentAlreadyExistException>();
        }
    }
}

File: exercise/java/day04/src/test/java/blog/ArticleTests.java
package blog;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

class ArticleTests {
    @Test
    void it_should_add_valid_comment() throws CommentAlreadyExistException {
        var article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        article.addComment("Amazing article !!!", "Pablo Escobar");
    }

    @Test
    void it_should_add_a_comment_with_the_given_text() throws CommentAlreadyExistException {
        var article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        var text = "Amazing article !!!";
        article.addComment(text, "Pablo Escobar");

        assertThat(article.getComments())
                .hasSize(1)
                .anyMatch(comment -> comment.text().equals(text));
    }

    @Test
    void it_should_add_a_comment_with_the_given_author() throws CommentAlreadyExistException {
        var article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        var author = "Pablo Escobar";
        article.addComment("Amazing article !!!", author);

        assertThat(article.getComments())
                .hasSize(1)
                .anyMatch(comment -> comment.author().equals(author));
    }

    @Test
    void it_should_add_a_comment_with_the_date_of_the_day() throws CommentAlreadyExistException {
        var article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        article.addComment("Amazing article !!!", "Pablo Escobar");
    }

    @Test
    void it_should_throw_an_exception_when_adding_existing_comment() throws CommentAlreadyExistException {
        var article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );
        article.addComment("Amazing article !!!", "Pablo Escobar");

        assertThatThrownBy(() -> {
            article.addComment("Amazing article !!!", "Pablo Escobar");
        }).isInstanceOf(CommentAlreadyExistException.class);
    }
}
